pkg/internal/testing/controlplane/apiserver.go

```diff
@@ -101,9 +147,51 @@ func (s *APIServer) Start() error {
 			return err
 		}
 	}
+	if err := s.Authn.Start(); err != nil {
+		return err
+	}
 	return s.processState.Start(s.Out, s.Err)
 }
 
+// configurePorts configures the serving ports for this API server.
+//
+// Most of this method currently deals with making the deprecated fields
+// take precedence over the new fields.
+func (s *APIServer) configurePorts() error {
+	// prefer the old fields to the new fields if a user set one,
+	// otherwise, default the new fields and populate the old ones.
+
+	// Insecure: URL, InsecureServing
+	if s.URL != nil {
+		s.InsecureServing = &process.ListenAddr{
+			Address: s.URL.Hostname(),
+			Port:    s.URL.Port(),
+		}
+	} else if insec := s.InsecureServing; insec != nil {
+		if insec.Port == "" || insec.Address == "" {
+			port, host, err := addr.Suggest("")
```

Comment:
Should we pass `InsecureServing.Address` down to `addr.Suggest`?
```suggestion
		if insec.Port == "" {
			port, host, err := addr.Suggest(insec.Address)
```

Reply:
I was mostly intending on porting the suggestion logic as-is from what it is currently to avoid breaking things -- the old code never passed anything to `addr.Suggest` AFAICT

Reply:
Yep, also checked how it was done before. This was just a suggestion on top, nothing critical.